Lane.cc

```diff
@@ -61,17 +61,17 @@ void Lane::processEventAsync(tbb::task_group& group, TaskHolder iCallback, const
   }
 }
 
-void Lane::doNextEvent(std::atomic<long>& index, tbb::task_group& group,  const OutputerBase& outputer, AtomicRefCounter counter) {
+void Lane::doNextEvent(std::atomic<long>& index, tbb::task_group& group,  const OutputerBase& outputer, TaskHolder finalTask) {
   using namespace std::string_literals;
   presentEventIndex_ = index++;
   if(source_->mayBeAbleToGoToEvent(presentEventIndex_)) {
     if(verbose_) {
       std::cout <<"event "+std::to_string(presentEventIndex_)+"\n"<<std::flush;
     }
     
-    OptionalTaskHolder processEventTask(group, make_functor_task([this,&index, &group, &outputer, counter]() {
-          TaskHolder recursiveTask(group, make_functor_task([this, &index, &group, &outputer, counter]() {
-                doNextEvent(index, group, outputer, std::move(counter));
+    OptionalTaskHolder processEventTask(group, make_functor_task([this,&index, &group, &outputer, finalTask]() {
```

Comment:
```suggestion
    OptionalTaskHolder processEventTask(group, make_functor_task([this,&index, &group, &outputer, finalTask=std::move(finalTask)]() {
```

Reply:
I assumed there was a reason the counter wasn't moved, but it seems ok